src/main/java/dk/bec/polonez/reservationsystem/service/ReservationService.java

```diff
@@ -100,9 +164,31 @@ public ResponseReservationDto updateReservation(UpdateReservationDto reservation
                 .build();
     }
 
-    public boolean deleteReservation(Long id) {
+    public boolean deleteReservation(Long id) throws ResponseStatusException {
+        User currentUser = authService.getCurrentUser();
+        Role currentRole = currentUser.getRole();
+
+        if (!currentRole.hasReservationDeletePrivilege())
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+
+        Reservation reservation = reservationRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.FORBIDDEN, "Error: This reservation is already approved!"));
+
+        if (!reservation.getStatus().equals(ReservationStatus.PENDING.name()))
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Error: This reservation is already approved!");
+
         reservationRepository.deleteById(id);
         return true;
     }
 
-}
+    private boolean isReservationCollidingWithOtherOnes(Reservation myReservation) {
+        List<Reservation> offerReservations = reservationRepository.findByOffer(myReservation.getOffer());
```

Comment:
tutaj wyciagacie wszystkie rezerwacje jakie kiedykolwiek powstaly? jesli to tak dziala, to po jakims czasie, bedzie to dzialac bardzo bardzo wolno.